File: Progetto-2023-2024-Vecchio-Larocca-Lentini-Mahrous/backend/src/main/java/it/unical/inf/ea/backend/dto/WishlistDTO.java
package it.unical.inf.ea.backend.dto;

import it.unical.inf.ea.backend.dto.enums.WishlistVisibility;
import lombok.*;

import java.time.LocalDateTime;
import java.util.List;

@Data
@NoArgsConstructor
@ToString
@Builder
@AllArgsConstructor
public class WishlistDTO {

    @NonNull
    private String id;

    @NonNull
    private String wishlistName;

    @NonNull
    private WishlistVisibility visibility;

    private List<ProductDTO> products;
}

File: frontend/app/src/main/java/com/android/frontend/dto/WishlistDTO.kt
package com.android.frontend.dto

import java.time.LocalDateTime

data class WishlistDTO (
    var id: String,
    var wishlistName: String,
    var wishlistVisibility: WishlistVisibility,
    var products: List<ProductDTO>? = null,
    var createdAt: LocalDateTime? = null
) {

    enum class WishlistVisibility{
        PUBLIC,
        PRIVATE,
        SHARED
    }
}



